symfony/plugins/orangehrmAttendancePlugin/Dao/AttendanceDao.php

```diff
@@ -41,120 +45,106 @@ public function savePunchRecord(AttendanceRecord $attendanceRecord) {
     }
 
     /**
-     * getLastPunchRecord
-     * @param $employeeId, $actionableStatesList
-     * @return AttendanceRecord
+     * @param int $employeeNumber
+     * @param string[] $actionableStatesList
+     * @return AttendanceRecord|null
      */
-    public function getLastPunchRecord($employeeId, $actionableStatesList) {
-
+    public function getLastPunchRecordByEmployeeNumberAndActionableList(int $employeeNumber, array $actionableStatesList): ?AttendanceRecord
+    {
+        $q = $this->createQueryBuilder(AttendanceRecord::class, 'attendanceRecord');
+        $q->andWhere('attendanceRecord.employee = :empNumber');
+        $q->andWhere($q->expr()->in('attendanceRecord.state', ':state'))
+            ->setParameter('state', $actionableStatesList);
+        $q->setParameter('empNumber', $employeeNumber);
+        $q->orderBy('attendanceRecord.id', ListSorter::DESCENDING);
```

Comment:
Will change `attendanceRecord.id` to handle `attendanceRecord.id`, `attendanceRecord.punchOutUtcTime` if applicable